src/Orckestra.Composer.Cart/Providers/Order/EditingOrderProvider.cs

```diff
@@ -53,6 +54,64 @@ public virtual async Task<bool> CanEdit(Overture.ServiceModel.Orders.Order order
             return isOrderEditable;
         }
 
+        public virtual async Task<bool> CanCancel(Overture.ServiceModel.Orders.Order order)
+        {
+            if (order?.Cart?.Shipments == null ||
+                OrderHistoryConfiguration.CompletedOrderStatuses.Contains(order.OrderStatus))
+            {
+                return false;
+            }
+            var orderFulfillmentState = await OrderRepository.GetOrderFulfillmentStateAsync(new GetOrderFulfillmentStateParam
+            {
+                OrderId = order.Id,
+                ScopeId = order.ScopeId
+            }).ConfigureAwait(false);
+
+            if (orderFulfillmentState.IsCancelable
+                && !orderFulfillmentState.IsProcessing
+                && !orderFulfillmentState.Status.Equals(Constants.OrderStatus.Canceled,
+                    StringComparison.InvariantCultureIgnoreCase)
+                && !orderFulfillmentState.Status.Equals("Completed", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public virtual async Task<bool> PendingCancel(Overture.ServiceModel.Orders.Order order)
+        {
+            if (order.OrderStatus.Equals(Constants.OrderStatus.Canceled, StringComparison.InvariantCultureIgnoreCase)) return false;
+
+            var orderFulfillmentState = await OrderRepository.GetOrderFulfillmentStateAsync(
+                new GetOrderFulfillmentStateParam
+                {
+                    OrderId = order.Id,
+                    ScopeId = order.ScopeId
+                }).ConfigureAwait(false);
+
+            return !orderFulfillmentState.IsCancelable && orderFulfillmentState.ShipmentFulfillmentStates.Any(item => item.AllowedStatusChanges.Contains(Constants.OrderStatus.Canceled));
```

Comment:
this ckeck about "AllowedStatusChanges"  should be in method CanCancel, no?
also need to make sure all shipments has this, and when we do Cancel we need execue Cancel for all Shipments